crates/bevy_render/src/lib.rs

```diff
@@ -114,27 +117,58 @@ struct ScratchRenderWorld(World);
 impl Plugin for RenderPlugin {
     /// Initializes the renderer, sets up the [`RenderStage`](RenderStage) and creates the rendering sub-app.
     fn build(&self, app: &mut App) {
+        app.add_asset::<Shader>()
+            .add_debug_asset::<Shader>()
+            .init_asset_loader::<ShaderLoader>()
+            .init_debug_asset_loader::<ShaderLoader>()
+            .register_type::<Color>();
+
         let options = app
             .world
             .get_resource::<settings::WgpuSettings>()
             .cloned()
             .unwrap_or_default();
 
-        app.add_asset::<Shader>()
-            .add_debug_asset::<Shader>()
-            .init_asset_loader::<ShaderLoader>()
-            .init_debug_asset_loader::<ShaderLoader>()
-            .register_type::<Color>();
+        bevy_utils::tracing::info!("Tryin to request primarywindow");
+
+        let mut system_state: SystemState<(
+            Query<&WindowHandle, With<Window>>,
+            Res<PrimaryWindow>,
+            // Res<WgpuSettings>,
+        )> = SystemState::new(&mut app.world);
+        let (
+            window_query,
+            primary_window,
+            // options, // This was .clone().unwrap_or_default(). Will this work the same?
+        ) = system_state.get(&mut app.world);
+
+        bevy_utils::tracing::info!("Should have resource here");
 
         if let Some(backends) = options.backends {
             let instance = wgpu::Instance::new(backends);
             let surface = {
-                let windows = app.world.resource_mut::<bevy_window::Windows>();
-                let raw_handle = windows.get_primary().map(|window| unsafe {
-                    let handle = window.raw_window_handle().get_handle();
-                    instance.create_surface(&handle)
-                });
-                raw_handle
+                // TODO: This can probably be cleaner
+                if let Ok(handle_component) = window_query.get(
+                    primary_window
+                        .window
+                        .expect("There should be a primary window"),
+                ) {
```

Comment:
Status update: Everything compiles now but panics here since the window-setup commands from `bevy_window` to create the primary-window has not been executed yet. 

Potential solutions would be to either special-case window-creation during build-time or to look into running commands during build-time.

Next step: Thanks to @mockersf, look into running commands directly on the world (in bevy_winit)